tests/pos-custom-args/qualified-types/tensorPostfixLambda.scala

```diff
@@ -0,0 +1,82 @@
+import language.experimental.postfixLambda
+
+
+type Shape = List[Int] with _.forall(_ > 0)
+
+extension (s: Shape)
+  //erased ?
+  def nbrElems = s.fold(1)(_ * _)
+  //erased ?
+  def reduce(axes: Shape) = s.zipWithIndex.filter((_, i) => axes.contains(i)).map(_._1)
```

Comment:
`axes` is a list of axis indices. Semantically, I don't think it should be a `Shape`, which are not list of axis indices, but of axis lengths.

Reply:
I can change it, I tried to stay as close as possible to https://mbovel.github.io/scalacon-typelevel-operations/presentation#/6/6

Reply:
I see, and I used that to stay close to https://github.com/MaximeKjaer/tf-dotty/blob/45af57dd0f60cb2d2fc9cf56f963b6ca4bd32909/modules/compiletime/src/main/scala/io/kjaer/compiletime/Shape.scala#L8-L31.

But as `Shape` is just a list of integers here, I would just make another type for indexes, like `type Indexes = List[Int]`.

Reply:
Hmm, what do you think of this:
```suggestion
  type Axes = List[Int] with _.forall(x => s.indicies.contains(x))
  def reduce(axes: Axes) = s.zipWithIndex.filter((_, i) => axes.contains(i)).map(_._1)
```

Reply:
Mmh why not, but that will be hard to check. Just a `type Indices = List[Int]` would be fine for now I think.

Typo: `indices`.

Reply:
```suggestion
  type Axis = Int with x => s.indicies.contains(x)
  def reduce(axes: List[Axis]) = s.zipWithIndex.filter((_, i) => axes.contains(i)).map(_._1)
```

Reply:
Looks good to me.